Tests/CSharp/SpikeTests.cs

```diff
@@ -0,0 +1,145 @@
+﻿using System;
+using System.Collections.Immutable;
+using System.IO;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Text.RegularExpressions;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Razor.Language;
+using Microsoft.CodeAnalysis;
+using Xunit;
+
+
+namespace ICSharpCode.CodeConverter.Tests.CSharp;
+
+public class SpikeTests
+{
+
+    [Fact]
+    public async Task ConvertRazorFileAsync()
+    {
+        var codeDocument = GetCodeDocument("""
+        @Code
+            ViewData("Title") = "About" & CStr(1)
+        End Code
+
+        <main aria-labelledby="title">
+            <h2 id="title">@ViewData("Title").</h2><h3 id="title">@ViewData("Title2").</h3>
+            <h3>@ViewData("Message")</h3>
+
+            <p>Use this area to provide additional information.</p>
+        </main>@Code
+            ViewData("Title") = "About" & CStr(1)
+        End Code
+
+        <main aria-labelledby="title">
+            <h2 id="title">@ViewData("Title").</h2><h3 id="title">@ViewData("Title2").</h3>
+            <h3>@ViewData("Message")</h3>
+
+            <p>Use this area to provide additional information.</p>
+        </main>
+
+        """, "anything.vbhtml");
+        
+        var syntaxTree = codeDocument.GetSyntaxTree();
+        var getRoot = syntaxTree.GetType().GetProperty("Root", BindingFlags.Instance | BindingFlags.NonPublic).GetMethod;
+        var root = getRoot.Invoke(syntaxTree, BindingFlags.Instance | BindingFlags.NonPublic, null, null, null);
+        var getDocument = root.GetType().GetProperty("Document", BindingFlags.Instance | BindingFlags.Public).GetMethod;
+        var document = getDocument.Invoke(root, BindingFlags.Instance | BindingFlags.NonPublic, null, null, null);
+        var getChildren = document.GetType().GetProperty("Children", BindingFlags.Instance | BindingFlags.Public).GetMethod;
+        var children = getChildren.Invoke(document, BindingFlags.Instance | BindingFlags.NonPublic, null, null, null);
```

Comment:
I'm pretty sure everything needed is in here, but it will take a lot of brittle reflection calls to internal methods to put it together.

Ultimately, need to get a vb syntaxtree for the whole file, and be able to map each code block back to the vbhtml so we can insert the relevant cs there at the end of conversion.